cpp/include/cudf/copying.hpp

```diff
@@ -64,7 +73,7 @@ namespace cudf {
 std::unique_ptr<table> gather(
   table_view const& source_table,
   column_view const& gather_map,
-  bool check_bounds                   = false,
+  out_of_bounds_policy bounds         = out_of_bounds_policy::NULLIFY,
```

Comment:
What we really want is to control what happens to negative indices. Do they wrap around or do they count as out of bounds?

Reply:
Wrap around (opt-in)

Reply:
Is the plan to add an additional parameter to control that behavior?

Reply:
We haven't discussed about it yet. Personally, I think we should remove the support of wrap-around for out-of-bound negative indices and only support for range `[-n_rows, n_rows)`. Thoughts?

Reply:
The previous logic makes sense to me. If the gather map is unsigned, then there aren't any negative values to wrap around.

Reply:
I'm OK with not exposing a parameter and implicitly using the signed-ness of the gathermap too. It doesn't matter for Python since we're _always_ going to need negative index handling.